internal/scheduler/scheduler.go

```diff
@@ -61,7 +63,11 @@ func Scheduler(apps []argov1alpha1.Application, stage syncv1alpha1.ProgressiveSy
 		return scheduledApps
 	}
 
-	for i := 0; i < maxParallel-len(progressingApps); i++ {
+	// It may be that at point when apps were observed, no apps were progressing yet
+	// so maxParallel-len(progressingApps) might actually be greater than len(outOfSyncApps)
+	// causing the runtime to panic
+	p := (int)(math.Min((float64)(maxParallel-len(progressingApps)), (float64)(len(outOfSyncApps))))
```

Comment:
I'd suggest clarifying this line with a if rather than the double casting. It's relatively simple to be explicited.
```
p := maxParallel-len(progressingApps)
if p > len(outofSyncApps){
  p = len(outOfSyncApps)
}
```